From: Stefan Behrens <sbehrens@giantdisaster.de>
Date: Thu, 26 Jul 2012 03:40:35 -0600
Patch-mainline: 3.6
Git-commit: aa2ffd06168e25689e0eb9662bf4595ba2bbac14
References: FATE#312888
Subject: [PATCH] Btrfs: fix a misplaced address operator in a
 condition

This should obviously not be "if (&flag)" but "if (flag)".

Signed-off-by: Stefan Behrens <sbehrens@giantdisaster.de>
Signed-off-by: David Sterba <dsterba@suse.cz>
---
 fs/btrfs/locking.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/fs/btrfs/locking.c
+++ b/fs/btrfs/locking.c
@@ -67,7 +67,7 @@ void btrfs_clear_lock_blocking_rw(struct
 {
 	if (eb->lock_nested) {
 		read_lock(&eb->lock);
-		if (&eb->lock_nested && current->pid == eb->lock_owner) {
+		if (eb->lock_nested && current->pid == eb->lock_owner) {
 			read_unlock(&eb->lock);
 			return;
 		}
